### PR TITLE
feat($parse): allow forcing expensive checks at the provider level

### PR DESCRIPTION
**Pull request by lgalfaso**
Allow forcing expensive checks at the $parse provider level

### CODE REVIEW

**Comment by Narretz** [comment]
Can you please add labels / milestone to the issue? And should the description be more explicitly what these expensive checks are and when you should enable them?

File: src/ng/parse.js [hunk 1]
```diff
@@ -1687,6 +1687,25 @@ function getValueOf(value) {
 function $ParseProvider() {
   var cacheDefault = createMap();
   var cacheExpensive = createMap();
+  var forceExpensiveChecks = false;
+
+  /**
+   * @ngdoc method
+   * @name $parseProvider#expensiveChecks
+   * @description
+   * Force expensive checks on all parsed expressions. Defaults to `false`.
```
Comment (gkalpak) [inline src/ng/parse.js]: I think a little explanation on what `false` means is in place (i.e. that will still apply expensive checks in some places or even what "expensive checks" include).

File: src/ng/parse.js [hunk 2]
```diff
@@ -1687,6 +1687,25 @@ function getValueOf(value) {
...
+   *
+   * @param {boolean=} value new value to set the force expensive checks to.
```
Comment (gkalpak) [inline src/ng/parse.js]: _force expensive checks_  -->  _forceExpensiveChecks_ ?

**Comment by lgalfaso** [comment]
Sorry for the noise, I thought that adding this option would help with another problem, but this is not the case. Closing

**Pull request closed without merging**